### DISCUSSION TITLE
Is it worth adding a note how to stand up docker local redis as well as docker local postgres?

### DISCUSSION THREAD

**Discussion by DaveParr** [loco-rs/loco#260] [General]
[Getting started tour](https://loco.rs/docs/getting-started/tour/) contains the note:

>  `docker run -d -p 5432:5432 -e POSTGRES_USER=loco -e POSTGRES_DB=loco_app -e POSTGRES_PASSWORD="loco" postgres:15.3-alpine`

Would it be worth extending it with a similar line for redis, such as:

``` sh
docker run -d -p 6379:6379 redis:alpine3.19
```

**Comment by kaplanelad** [comment]
Thank you, @DaveParr. Each starter has a Docker Compose file that simplifies running PostgreSQL and Redis. You can find it [here](https://github.com/loco-rs/loco/blob/master/starters/rest-api/.devcontainer/docker-compose.yml).

Would you be interested in opening a pull request to incorporate this addition?

**Reply by DaveParr** [reply]
I would be 😄 . Having a pass at playing with the linked `docker-compose.yml` now.

**Reply by DaveParr** [reply]
@kaplanelad I've got a few questions having played with the `docker-compose.yml` file.

Am I right that this is not intended to be either 
1) part of the output of the starter?
2) part of the result of running `cargo loco generate deployment` and selecting `docker`?

because right now on my current version of loco (`loco-cli 0.2.5`) I've not spotted these docker files in my local codebase, so I've copied them manually from GH into my SaaS starter project.

I've attempted to see if I can get a running local development environment from using the `docker-compose.yml` you linked, along with the `.env` in the same directory on GH and the `Dockerfile` on GH. 

The outcome is that the docker environments are all running without error, but I'm not able to open a browser to interact with the app. The `docker compose ps` reports the port as 'sleep infinity'.

What is the intent of running this `sleep infinity` command ? (I don't have much experience with vscode .devcontainers, so I'm assuming that's why this is a little alient to me

In prior work on django I've commonly set this to be the equivalent to the `loco start` command. I've tried running things like `docker compose run test_loco_auth-app loco start` (using `docker compose ps`  to find the container name) but this has errored with: `no such service: test_loco_auth-app`.

Either way I've opened #263 as maybe that's all that's needed for most people to solve this, but I'm still a little confused 🥲 . I have to put this away for now and get on with my day, but I'll try and come poke at this specific issue next time I have a chance. For now I'm assuming that I need to go poke the devcontainer docs for VS code next to understand exactly what I'm handling.

Hope I'm being slightly more helpful than I am annoying!

**Reply by kaplanelad** [reply]
@DaveParr `loco-cli` is only for starters, and the deployment generator is unavailable there. 

When you start a new project with `loco new`, the started folder that was created contains the `cargo loco generate deployment.`.

if you don't want to use `.devcontainer` you can run this command `docker-compose -f ./.devcontainer/docker-compose.yml up` in your project
